aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,201 @@
+    return GlueClient.builder()
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  @Override
+  public KmsClient kms() {
+    return KmsClient.builder()
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  @Override
+  public DynamoDbClient dynamo() {
+    return DynamoDbClient.builder()
+        .httpClientBuilder(UrlConnectionHttpClient.builder())
+        .build();
+  }
+
+  private AwsCredentialsProvider buildS3CredentialProvider() {
+
+    GlueClient glue = glue();
+    String glueCatalogId = properties.get(AwsProperties.GLUE_CATALOG_ID);
+    String databaseName = properties.get(AwsProperties.LAKE_FORMATION_DB_NAME);
+    String tableName = properties.get(AwsProperties.LAKE_FORMATION_TABLE_NAME);
+
+    GetTableResponse response = glue.getTable(GetTableRequest.builder()
+        .catalogId(glueCatalogId)
+        .databaseName(databaseName)
+        .name(tableName)
+        .build());
+    boolean isRegisteredWithLakeFormation = response.table().isRegisteredWithLakeFormation() != null ?
+        response.table().isRegisteredWithLakeFormation() : Boolean.FALSE;
+
+    if (isRegisteredWithLakeFormation) {
+      return new LakeFormationCredentialsProvider(properties);
+    } else {
+      return LakeFormationCredentialsProvider.getAssumeRoleCredentialsProvider(properties);
+    }
+  }
+
+  private static Region region(Map<String, String> properties) {
+    String region = PropertyUtil.propertyAsString(properties, AwsProperties.CLIENT_ASSUME_ROLE_REGION, null);
+    if (region == null || region.isEmpty()) {
+      throw new RuntimeException("Region is empty");
+    }
+    return Region.of(region);
+  }
+
+  static class LakeFormationCredentialsProvider implements AwsCredentialsProvider {
+    private LakeFormationClient client;
+    private String tableArn;
+
+    LakeFormationCredentialsProvider(Map<String, String> properties) {
+      AwsCredentialsProvider awsCredentialsProvider = getAssumeRoleCredentialsProvider(properties);
+      String region = PropertyUtil.propertyAsString(properties, AwsProperties.CLIENT_ASSUME_ROLE_REGION, null);
+      client = buildLakeFormationClient(awsCredentialsProvider, region);
+      tableArn = getTableArn(properties);
+    }
+
+    @Override
+    public AwsCredentials resolveCredentials() {
+      GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
+          GetTemporaryGlueTableCredentialsRequest.builder()
+              .tableArn(tableArn)
+              .permissions(Permission.ALL)
+              .build();
+      GetTemporaryGlueTableCredentialsResponse response =
+          client.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+      return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+    }
+
+    private static AwsCredentialsProvider getAssumeRoleCredentialsProvider(Map<String, String> properties) {
```

Comment:
prefer to not use `getXXX`, either just name the method as `assumeRoleCredentialsProvider`, or use meaningful verbs such as `buildAssumeRoleCredentialsProvider`